sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -217,9 +222,16 @@ private[sql] case class DataSourceScan(
     // TODO: The abstractions between this class and SqlNewHadoopRDD makes it difficult to know
     // here which path to use. Fix this.
 
-    ctx.INPUT_ROW = row
+    // These assignments avoids accesses to instance variables in a while-loop
+    // Since they are loop invariant variables (LIVs), host them outside the while-loop
+    val colLocalVars = output.zipWithIndex.map(x => ctx.freshName("col" + x._2))
+    val columnLIVAssigns = (colLocalVars zip colVars).map { case (localName, name) =>
```

Comment:
An access to an instance variable in ```colVars```, which requires ```getField``` bytecode, would involve memory access. We expect that an access to a local variable in ```colLocalVars```, which requires ```aload``` bytecode, would use register access by a Java JIT compiler. 
This is why we introduced ```colLocalVars```.